packages/scene-composer/src/components/panels/scene-components/common/ValueDataBindingBuilder.tsx

```diff
@@ -85,19 +89,13 @@ export const ValueDataBindingBuilder: React.FC<IValueDataBindingBuilderProps> =
       setBuilderState(filterBuilderState(state));
     });
     return () => valueDataBindingStore.setOnStateChangedListener(undefined);
-  }, [valueDataBindingProvider]);
+  }, [valueDataBindingStore]);
 
   useEffect(() => {
     // Initiate the provider
     const state = valueDataBindingStore.setBinding(componentRef, binding, dataBindingConfig);
-    setBuilderState(filterBuilderState(state));
-  }, [binding, valueDataBindingProvider, dataBindingConfig]);
-
-  useEffect(() => {
-    const state = valueDataBindingStore.setBinding(componentRef, binding, dataBindingConfig);
-    setBuilderState(filterBuilderState(state));
     setAutoSuggestValue(state.selectedOptions[ENTITY_ID_INDEX]?.value || '');
-  }, [componentRef]);
+  }, [componentRef, binding, valueDataBindingStore, dataBindingConfig]);
```

Comment:
This may be an area which would benefit from refactoring soon. A large dependency list is something to be wary of.

Reply:
logically their changes should all trigger the binding and autoSuggestValue change

Reply:
this auto suggest value can be set from two sources, one is from the `Autosuggest` component onChange (which is from user input), and another is from data binding change caused by events like undo/redo scene update, selecting a different node, etc. so it needs to be in state

Reply:
data binding change caused by events like undo/redo is not when user start typing. we don't save the value to store until its a valid id, so there will be a value from store and a value from local state